Intersect.Client/Entities/Player.cs

```diff
@@ -2251,125 +2251,124 @@ public virtual void DrawGuildName(Color textColor, Color borderColor = null, Col
-                                    {
-                                        // Skip if Friend is GuildMate.
-                                        if (Globals.Database.GuildMemberOverheadInfo && isGuildMate)
-                                        {
-                                            continue;
-                                        }
+                // If PartyMemberOverheadInfo is toggled off, draw party members overhead information.
+                else if (!Globals.Database.PartyMemberOverheadInfo && isPartyMate)
+                {
+                    en.Value.DrawName(null);
+                }
 
-                                        en.Value.DrawName(null);
-                                    }
+                // If FriendOverheadInfo & GuildMemberOverheadInfo are off: let's prevent double draw / overlapping.
+                else if (!Globals.Database.FriendOverheadInfo && isFriend &&
+                         !Globals.Database.GuildMemberOverheadInfo && isGuildMate && !isPartyMate)
+                {
+                    en.Value.DrawName(null);
+                }
 
-                                    // If GuildMemberOverheadInfo is toggled off, draw Guild Members
-                                    // overhead information only when hovered by the cursor.
-                                    if (!Globals.Database.GuildMemberOverheadInfo && isGuildMate && !isPartyMate)
-                                    {
-                                        // Skip if GuildMate is Friend.
-                                        if (Globals.Database.FriendOverheadInfo && isFriend)
-                                        {
-                                            continue;
-                                        }
+                // If FriendOverheadInfo is toggled off, draw friends overhead information.
+                else if (!Globals.Database.FriendOverheadInfo && isFriend && !isPartyMate)
+                {
+                    // Skip if friend is a guild mate.
+                    if (Globals.Database.GuildMemberOverheadInfo && isGuildMate)
+                    {
+                        continue;
+                    }
 
-                                        en.Value.DrawName(null);
-                                    }
-                                }
-                            }
-                        }
+                    en.Value.DrawName(null);
+                }
 
-                        break;
+                // If GuildMemberOverheadInfo is toggled off, draw guild members overhead information.
+                else if (!Globals.Database.GuildMemberOverheadInfo && isGuildMate && !isPartyMate)
+                {
+                    // Skip if guild mate is a friend.
+                    if (Globals.Database.FriendOverheadInfo && isFriend)
+                    {
+                        continue;
                     }
+
+                    en.Value.DrawName(null);
                 }
```

Comment:
I think all of this logic needs to be looked at with a fine-toothed comb and simplified, as well as detailed explanations if things are double-checked.